github/repos_environments.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2021 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// Environment represents a single environment in a repository.
+type Environment struct {
+	Owner                  *string         `json:"owner,omitempty"`
+	Repo                   *string         `json:"repo,omitempty"`
+	EnvironmentName        *string         `json:"environment_name,omitempty"`
+	WaitTimer              *int            `json:"wait_timer,omitempty"`
+	Reviewers              []*EnvReviewers `json:"reviewers,omitempty"`
+	DeploymentBranchPolicy *BranchPolicy   `json:"deployment_branch_policy,omitempty"`
```

Comment:
I wonder if I can make a comment here for some clarification on what these fields actually mean. The fields below cover the response from the API for this object, and afaict these fields are always nil.

I am sure I am missing something, so if possible could someone explain what they are for?

Reply:
The [official docs](https://docs.github.com/en/rest/deployments/environments?apiVersion=2022-11-28#get-an-environment) say this:

> Note: To get information about name patterns that branches must match in order to deploy to this environment, see "[Get a deployment branch policy](https://docs.github.com/rest/deployments/branch-policies#get-a-deployment-branch-policy)."

Does that help?  
Do you want to create a PR explaining your understanding of the official docs to this repo?

Reply:
Hi guys, thanks for the response!

I am referring to the highlighted vales (`Owner` `Repo` `EnvironmentName` `WaitTimer` `Reviewers` `DeploymentBranchPolicy`). I am seeing only nil values in `GET` operations (this struct is only used in `GET` operations at this point --> `ListEnvironments()` & `GetEnvironment)_`, we are using the `CreateUpdateEnvironment` struct for `PUT` operations with `CreateUpdateEnvironment()`) and these values do not appear in the [docs](https://docs.github.com/en/rest/deployments/environments?apiVersion=2022-11-28).

Again, I might be missing something. My use case is that I want the `Reviewers` value, and at this point I am parsing the `ProtectionRules` in my code. Was the idea that we parse this slice of rules (and populate the other fields) as we unmarshal?